src/containers/audio-rewards-page/components/modals/TrendingRewards.tsx

```diff
@@ -0,0 +1,168 @@
+const useRewardsType = (): [
+  TrendingRewardsModalType,
+  (type: TrendingRewardsModalType) => void
+] => {
+  const dispatch = useDispatch()
+  const rewardsType = useSelector(getTrendingRewardsModalType)
+  const setTrendingRewardsType = useCallback(
+    (type: TrendingRewardsModalType) => {
+      dispatch(setTrendingRewardsModalType({ modalType: type }))
+    },
+    [dispatch]
+  )
+  return [rewardsType, setTrendingRewardsType]
+}
+
+const useTweetId = (type: TrendingRewardsModalType) => {
+  const tracksId = useRemoteVar(StringKeys.REWARDS_TWEET_ID_TRACKS)
+  const playlistsId = useRemoteVar(StringKeys.REWARDS_TWEET_ID_PLAYLISTS)
+  return type === 'tracks' ? tracksId : playlistsId
+}
+
+const TrendingRewardsBody = ({
+  dismissModal
+}: {
+  dismissModal: () => void
+}) => {
+  const [modalType, setModalType] = useRewardsType()
+
+  const tabOptions = [
+    {
+      key: 'tracks',
+      text: messages.topTracks
+    },
+    {
+      key: 'playlists',
+      text: messages.topPlaylists
+    }
+  ]
+
+  const navigate = useNavigateToPage()
+
+  const onButtonClick = useCallback(() => {
+    const page =
+      modalType === 'tracks' ? TRENDING_PAGE : TRENDING_PLAYLISTS_PAGE
+    navigate(page)
+    dismissModal()
+  }, [navigate, modalType, dismissModal])
+
+  const wm = useWithMobileStyle(styles.mobile)
+
+  // If we change type, show the spinner again
+  const [showSpinner, setShowSpinner] = useState(true)
+  useEffect(() => {
+    setShowSpinner(true)
+  }, [modalType])
+
+  const tweetId = useTweetId(modalType)
+
+  return (
+    <div className={wm(styles.container)}>
+      <div className={styles.sliderContainer}>
+        <TabSlider
+          options={tabOptions}
+          selected={modalType}
+          onSelectOption={option =>
+            setModalType(option as TrendingRewardsModalType)
+          }
+          textClassName={styles.slider}
+          activeTextClassName={styles.activeSlider}
+        />
+      </div>
+      <div className={styles.titles}>
+        <span className={styles.title}>{textMap[modalType].title}</span>
+        <span className={styles.subtitle}>{messages.winners}</span>
+      </div>
+      <div className={styles.insetRegion}>
+        <span className={styles.lastWeek}>{messages.lastWeek}</span>
+        <div className={styles.embedWrapper}>
+          {showSpinner && <LoadingSpinner className={styles.spinner} />}
+          <TwitterTweetEmbed
+            // Refresh it when we toggle
+            key={`twitter-${modalType}`}
+            tweetId={tweetId}
+            onLoad={() => setShowSpinner(false)}
+            options={{
+              conversation: 'none',
+              hide_thread: true,
+              width: 550,
+              height: 390
+            }}
+          />
+        </div>
+      </div>
+      <ButtonWithArrow
+        text={textMap[modalType].button}
+        onClick={onButtonClick}
+        className={styles.button}
+      />
+      <a href='' className={styles.terms}>
```

Comment:
Is this supposed to be a link?

Reply:
yeah broken for now, gonna add on next iteration